src/builder.rs

```diff
@@ -151,8 +182,15 @@ impl ActionInfo {
         let ak: SpendValidatingKey = self.spend.fvk.clone().into();
         let alpha = pallas::Scalar::random(&mut rng);
         let rk = ak.randomize(&alpha);
+        let note_type = self.spend.note.note_type();
```

Comment:
This is unnecessary since we asserted that it's the same as `self.output.note_type` assigned above.
```suggestion
```